zkvm/src/predicate.rs

```diff
@@ -71,6 +73,39 @@ impl Predicate {
     }
 }
 
+impl PredicateWitness {
+    pub fn to_point(&self) -> CompressedRistretto {
+        self.to_uncompressed_point().compress()
+    }
+
+    fn to_uncompressed_point(&self) -> RistrettoPoint {
+        let gens = PedersenGens::default();
+        match self {
+            PredicateWitness::Key(s) => s * gens.B,
+            PredicateWitness::Or(b) => {
```

Comment:
why is it `b`? :-) Also: you can match immediately on a nested tuple:

```suggestion
            PredicateWitness::Or((l, r)) => {
```

...

Reply:
(now i see why it's `b` - it's "boxed" pair - see the comment below)